### PR TITLE
NAS-109209 / 21.06 / Directory Service redesign work

### PR DESCRIPTION
**Pull request by AlexKarpov98**

### CODE REVIEW

**Comment by denysbutenko** [comment]
image
Can we somehow improve displaying empty values for tables?

File: src/app/helptext/directoryservice/dashboard.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        title: T('Idmap'),
+    },
+    ldap: {
+        title: T('LDAP'),
+        hostname: T('Hostname'),
+        baseDN: T('Base DN'),
+        bindDN: T('Bind DN'),
+    },
+    kerberosSettings: {
+        title: T('Kerboros Settings'),
```
Comment (denysbutenko) [inline src/app/helptext/directoryservice/dashboard.ts:20]: It seems like a typo

File: src/app/helptext/directoryservice/dashboard.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        appdefaults: T('Appdefaults Auxiliary Parameters'),
+        libdefaults: T('Libdefaults Auxiliary Parameters'),
+    },
+    kerboerosRealms: {
+        title: T('Kerboeros Realms'),
```
Comment (denysbutenko) [inline src/app/helptext/directoryservice/dashboard.ts:25]: same as above

File: src/app/helptext/directoryservice/dashboard.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    },
+    kerboerosKeytab: {
+        title: T('Kerboeros Keytab'),
```
Comment (denysbutenko) [inline src/app/helptext/directoryservice/dashboard.ts:28]: same as above

**Comment by denysbutenko** [comment]
I wondered if it reasonable to use wide forms for `Idmap`, `Kerberos Settings`, `Kerberos Realm`, and `Kerberos Keytab`?
An example of what I mean
Before:
image
After:
image

**Comment by AlexKarpov98** [comment]
@denysbutenko Fixed the issues you mentioned. Please check them.

**Comment by denysbutenko** [comment]
Looks much better now! Thank you, Alex

File: src/app/pages/directoryservice/idmap/idmap-form.component.ts [hunk 4]
```diff
@@ -1,34 +1,35 @@
...
+  protected title: string;
   protected isEntity: boolean = true;
   protected namesInUse = [];
   protected queryCall = 'idmap.query';
   protected addCall = 'idmap.create';
   protected editCall = 'idmap.update';
-  protected queryCallOption: Array<any> = [["id", "="]];
+  protected pk: any;
+  protected queryKey = 'id';
+  private getRow = new Subscription;
```
Comment (denysbutenko) [inline src/app/pages/directoryservice/idmap/idmap-form.component.ts:25]: I wonder if this shouldn't just have the `Subscription` type (seen in other places too)?

**Review by denysbutenko — APPROVED** [review]

**Pull request merged**